calendar-bundle/tests/EventListener/BreadcrumbListenerTest.php

```diff
@@ -0,0 +1,384 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\Tests\EventListener;
+
+use Contao\CalendarEventsModel;
+use Contao\CalendarModel;
+use Contao\Config;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Events;
+use Contao\Input;
+use Contao\CalendarBundle\EventListener\BreadcrumbListener;
+use Contao\PageModel;
+use Contao\TestCase\ContaoTestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+
+final class BreadcrumbListenerTest extends ContaoTestCase
+{
+    private const ROOT_PAGE_ID = 1;
```

Comment:
I don't think constants in unit tests are a good practice.

Reply:
I like it make it explicit which values are assumed as constant values in the unit tests. Anyway, removed in 24b7df3.